### PR TITLE
Implement `filter_permitted_menu_items` in AWS auth manager

### PR DESCRIPTION
**Pull request by vincbeck**
Implement `filter_permitted_menu_items` in AWS auth manager. This allows optimization on menu generation. Instead of making one call for each menu item, we do one for the entire menu.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/auth_manager/avp/facade.py [hunk 1]
```diff
@@ -125,62 +130,97 @@ def is_authorized(
...
 
-    def batch_is_authorized(
+    def get_batch_is_authorized_results(
         self,
         *,
         requests: Sequence[IsAuthorizedRequest],
-        user: AwsAuthManagerUser | None,
-    ) -> bool:
+        user: AwsAuthManagerUser,
+    ):
```
Comment (o-nikolas) [inline airflow/providers/amazon/aws/auth_manager/avp/facade.py]: missing return type annotation

File: airflow/providers/amazon/aws/auth_manager/avp/facade.py [hunk 2]
```diff
@@ -205,3 +245,25 @@ def _build_context(context: dict | None) -> dict | None:
         return {
             "contextMap": context,
         }
+
+    def _build_is_authorized_request_payload(self, request: IsAuthorizedRequest, user: AwsAuthManagerUser):
+        """
+        Build payload of individual authorization request send through ``batch_is_authorized`` API.
```
Comment (o-nikolas) [inline airflow/providers/amazon/aws/auth_manager/avp/facade.py]: This is what I think this is trying to say, is that right?

```suggestion
        Build a payload of an individual authorization request that could be sent through the ``batch_is_authorized`` API.
```

**Review by o-nikolas — APPROVED** [review]
Just some minor comments

File: airflow/auth/managers/base_auth_manager.py [hunk 3]
```diff
@@ -367,7 +367,7 @@ def _is_permitted_dag_id(method: ResourceMethod, methods: Container[ResourceMeth
             if _is_permitted_dag_id("GET", methods, dag_id) or _is_permitted_dag_id("PUT", methods, dag_id)
         }
 
-    def get_permitted_menu_items(self, menu_items: list[MenuItem]) -> list[MenuItem]:
+    def filter_permitted_menu_items(self, menu_items: list[MenuItem]) -> list[MenuItem]:
```
Comment (potiuk) [inline airflow/auth/managers/base_auth_manager.py:370]: Isn't that backwards-incompatible? Or we do not care too much at this stage?

File: airflow/providers/amazon/aws/auth_manager/avp/facade.py (hunk 2 above)
Comment (vincbeck) [reply airflow/providers/amazon/aws/auth_manager/avp/facade.py]: Correct, thanks :)

File: airflow/auth/managers/base_auth_manager.py (hunk 3 above)
Comment (vincbeck) [reply airflow/auth/managers/base_auth_manager.py:370]: You're right. I dont think anybody use that interface yet so I thought it was the good time to do it. I guess your statement "Or we do not care too much at this stage?" is correct :)

Comment (potiuk) [reply airflow/auth/managers/base_auth_manager.py:370]: Fine for me. I **think** we marked the AuthManager interface as experimental as well? In which case it's perfectly fine. Though I think that maybe a newsfragment (not significant - one liner) would be appropriate to add in this case.

Comment (vincbeck) [reply airflow/auth/managers/base_auth_manager.py:370]: I can do that :)

Comment (vincbeck) [reply airflow/auth/managers/base_auth_manager.py:370]: Done

**Review by potiuk — APPROVED** [review]

**Pull request merged**